source/framework/vx_image.c

```diff
@@ -376,6 +383,313 @@ static vx_status ownAllocImageBuffer(vx_reference ref)
+                break;
+            }
+        }
+        else
+        {
+            tivxCheckStatus(&status, vxMapImagePatch(input, &rect, i, &map_id, &addr, &ptr, (vx_enum)VX_READ_ONLY, (vx_enum)VX_MEMORY_TYPE_HOST, 0));
+            if ((vx_status)VX_SUCCESS == status)
+            {
+                status = vxCopyImagePatch(output, &rect, i, &addr, ptr, (vx_enum)VX_WRITE_ONLY, (vx_enum)VX_MEMORY_TYPE_HOST);
+            }
+            tivxCheckStatus(&status, vxUnmapImagePatch(input, map_id));
+        }
+    }
+    return status;
+}
+
+/*! \brief Adjust the memory pointer of the tensor or image by the given offset
+ * This involves non-recursively visiting every sub-image and sub-tensor
+ * and adjusting the memory pointer for that, as well.
+ */
+static vx_status adjustMemoryPointer(vx_image ref, uint64_t offset[TIVX_IMAGE_MAX_PLANES])
+{
+    vx_status status = (vx_status)VX_SUCCESS;
+    /* the stack can hold two time the max number of subimages*/
+    vx_image stack[TIVX_SUBIMAGE_STACK_SIZE];
+    vx_image local_img;
+    vx_image *subimages = NULL;
+    vx_uint32 stack_pointer = 0;
+    vx_uint32 i;
+
+    for (i = 0; i < TIVX_IMAGE_MAX_SUBIMAGES; ++i)
+    {
+        stack[i] = NULL;
+    }
+    stack[stack_pointer] = ref;
+    stack_pointer++;
+    while (0U < stack_pointer)
+    {
+        stack_pointer--;
+        local_img = stack[stack_pointer];
+        subimages = local_img->subimages;
+        tivx_obj_desc_image_t *obj_desc = (tivx_obj_desc_image_t *)local_img->base.obj_desc;
+        if (obj_desc->planes > 0U)
+        {
+            for (i = 0; i < obj_desc->planes; ++i)
+            {
+                obj_desc->mem_ptr[i].host_ptr = obj_desc->mem_ptr[i].host_ptr + offset[i];
+                obj_desc->mem_ptr[i].shared_ptr = tivxMemHost2SharedPtr(obj_desc->mem_ptr[i].host_ptr, (vx_enum)TIVX_MEM_EXTERNAL);
+            }
+        }
+        else
+        {
+            obj_desc->mem_ptr[0U].host_ptr = obj_desc->mem_ptr[0].host_ptr + offset[local_img->channel_plane];
+            obj_desc->mem_ptr[0U].shared_ptr = tivxMemHost2SharedPtr(obj_desc->mem_ptr[0U].host_ptr, (vx_enum)TIVX_MEM_EXTERNAL);
+        }
+        for (i = 0; i < TIVX_IMAGE_MAX_SUBIMAGES; ++i)
+        {
+            if (NULL != subimages[i])
+            {
+                if (TIVX_SUBIMAGE_STACK_SIZE < stack_pointer)
+                {
+                    VX_PRINT(VX_ZONE_ERROR, "Too many sub-images, may need to increase the value of TIVX_SUBIMAGE_STACK_SIZE\n");
+                    status = (vx_status)VX_ERROR_NO_RESOURCES;
+                    break;
+                }
+                else
+                {
+                    stack[stack_pointer] = subimages[i];
+                    stack_pointer++;
+                }
+            }
+        }
+    }
+    return status;
+}
+
+static vx_status swapImage(const vx_image input, const vx_image output)
+{
+    /* Swap image handles. Need to recalculate for all the sub-images
+        and sub-tensors.
```

Comment:
you mention sub-tensors here in the comment, but I think this is only for sub-images, correct?

Reply:
yes we had previously sub-tensor, but it was removed because it belongs to another extension.
comment removed with the latest commit.